Enable predict function to generate only one prediction when uncertainty is not needed, for speed up purpose.

shirleyuw opened the pull request:
The motivation is to speed up prediction when running on a list of benchmark datasets. 
The predict() function in Forecaster() class used to always predict 100 samples and return mean, quantiles and samples. To speed up, we only predict one sample when the uncertainty argument in Forecaster() is False, when uncertainty is not needed by the user.

bischtob commented on deep4cast/forecasters.py:
```diff
@@ -188,18 +193,30 @@ def predict(self, data, n_samples=100, quantiles=(5, 95)):
+        if not self.uncertainty:
+            # Return only mean prediction based on one sample when Uncertainty
+            # is False
+            print("Uncertainty is set to False. Only mean prediction is"
```
Please don't use a print statement for alerting the user here. Instead set the other statics to 'None' or 'NaN' when there is only one sample, so that code that depends on the presence of these fields downstream can simply handle this via testing for Nones or NaNs.

I also suggest shortening this part of the code by using inline ifs. Example:
``lower_quantile = np.nanpercentile(samples, quantiles[0], axis=0) if self.uncertainty else None``

shirleyuw commented:
Changed to inline ifs;
Changed to always return the same set of variables. When uncertainty is not required, it returns None for the quantile results.
